pkg/util/jaeger.go

```diff
@@ -0,0 +1,40 @@
+package util
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	jaeger "github.com/uber/jaeger-client-go"
+	jaegercfg "github.com/uber/jaeger-client-go/config"
+)
+
+type nopCloser struct {
+}
+
+func (nopCloser) Close() error { return nil }
+
+// TraceInit registers Jaeger as the OpenTracing implementation.
+func TraceInit(serviceName string) io.Closer {
+	jaegerAgentHost := os.Getenv("JAEGER_AGENT_HOST")
```

Comment:
We use command-line flags rather than environment variables for everything else. Why use an env var here?

Reply:
The environment variable, when distributor is run on kubernetes, allows to use in configs constructs like:
```
env:
- name: JAEGER_AGENT_HOST
   valueFrom:
     fieldRef:
       fieldPath: status.hostIP
```
See https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/ .

It's possible to achieve the same result using command-line arguments by using files, but it's slightly more complicated.
See https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/

This setup allows to easily run one jaeger agent per host instead of one per pod.